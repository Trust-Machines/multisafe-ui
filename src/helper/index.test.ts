import {
    formatUnits,
    parseUnits,
    checkDecimalAmount,
    checkAmount,
    detectTransactionType,
    contractPrincipalCVFromString,
    transformNftUri,
    isValidRecipient
} from './';
import {DEPLOYERS} from '@trustmachines/multisafe-contracts';

test('1 parseUnits', () => {
    expect(parseUnits("1.256", 6)).toMatchSnapshot();
    expect(parseUnits("120000", 6)).toMatchSnapshot();
    expect(parseUnits("1.256", 8)).toMatchSnapshot();
    expect(parseUnits("11", 0)).toMatchSnapshot();
});

test('2 formatUnits', () => {
    expect(formatUnits("1256000", 6)).toMatchSnapshot();
    expect(formatUnits("120000000000", 6)).toMatchSnapshot();
    expect(formatUnits("125600000", 8)).toMatchSnapshot();
    expect(formatUnits("11", 0)).toMatchSnapshot();
});

test('3 checkDecimalAmount', () => {
    expect(checkDecimalAmount(" ")).toMatchSnapshot();
    expect(checkDecimalAmount("0.1")).toMatchSnapshot();
    expect(checkDecimalAmount("43")).toMatchSnapshot();
    expect(checkDecimalAmount("100.23")).toMatchSnapshot();
    expect(checkDecimalAmount("100.123456")).toMatchSnapshot();
    expect(checkDecimalAmount("100.1234567")).toMatchSnapshot();
});

test('4 checkAmount', () => {
    expect(checkAmount("100")).toMatchSnapshot();
    expect(checkAmount("1.1")).toMatchSnapshot();
});

test('5 detectTransactionType', () => {
    expect(detectTransactionType(`${DEPLOYERS[0]}.add-owner`)).toMatchSnapshot();
    expect(detectTransactionType(`SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.add-owner`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.remove-owner`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.set-threshold`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.transfer-stx`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.transfer-sip-009`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.transfer-sip-010`)).toMatchSnapshot();
    expect(detectTransactionType(`${DEPLOYERS[0]}.transfer-token`)).toMatchSnapshot();
});

test('6 contractPrincipalCVFromString', () => {
    expect(contractPrincipalCVFromString(`SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.add-owner`)).toMatchSnapshot();
});

test('7 transformNftUri', () => {
    expect(transformNftUri('ipfs://QmYTX3u58v2Ero2drdtqhL6rPE5qnv51EJZ6WSu3LKqUBN/crashpunks-{id}.json', '3')).toMatchSnapshot();
    expect(transformNftUri('ar://II4z2ziYyqG7-kWDa98lWGfjxRdYOx9Zdld9P_I_kzE/3.json', '15')).toMatchSnapshot();
    expect(transformNftUri('https://api.satoshibles.com/token/btc/{id}', '25')).toMatchSnapshot();
});

test('8 isValidRecipient', () => {
    expect(isValidRecipient('')).toMatchSnapshot();
    expect(isValidRecipient('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7')).toMatchSnapshot();
    expect(isValidRecipient('SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.contract')).toMatchSnapshot();
    expect(isValidRecipient('foo')).toMatchSnapshot();
});