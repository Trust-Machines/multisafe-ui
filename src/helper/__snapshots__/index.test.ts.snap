// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 parseUnits 1`] = `"1256000"`;

exports[`1 parseUnits 2`] = `"120000000000"`;

exports[`1 parseUnits 3`] = `"125600000"`;

exports[`1 parseUnits 4`] = `"11"`;

exports[`2 formatUnits 1`] = `"1.256"`;

exports[`2 formatUnits 2`] = `"120000"`;

exports[`2 formatUnits 3`] = `"1.256"`;

exports[`2 formatUnits 4`] = `"11"`;

exports[`3 checkDecimalAmount 1`] = `false`;

exports[`3 checkDecimalAmount 2`] = `true`;

exports[`3 checkDecimalAmount 3`] = `true`;

exports[`3 checkDecimalAmount 4`] = `true`;

exports[`3 checkDecimalAmount 5`] = `true`;

exports[`3 checkDecimalAmount 6`] = `false`;

exports[`4 checkAmount 1`] = `true`;

exports[`4 checkAmount 2`] = `false`;

exports[`5 detectTransactionType 1`] = `"add-owner"`;

exports[`5 detectTransactionType 2`] = `null`;

exports[`5 detectTransactionType 3`] = `"remove-owner"`;

exports[`5 detectTransactionType 4`] = `"set-threshold"`;

exports[`5 detectTransactionType 5`] = `"transfer-stx"`;

exports[`5 detectTransactionType 6`] = `"transfer-sip-009"`;

exports[`5 detectTransactionType 7`] = `"transfer-sip-010"`;

exports[`5 detectTransactionType 8`] = `null`;

exports[`6 contractPrincipalCVFromString 1`] = `
Object {
  "address": Object {
    "hash160": "fad413a3b38e9cc64afb1bc0ecae4b948966785d",
    "type": 0,
    "version": 22,
  },
  "contractName": Object {
    "content": "add-owner",
    "lengthPrefixBytes": 1,
    "maxLengthBytes": 128,
    "type": 2,
  },
  "type": 6,
}
`;

exports[`7 transformNftUri 1`] = `"https://gateway.ipfs.io/ipfs/QmYTX3u58v2Ero2drdtqhL6rPE5qnv51EJZ6WSu3LKqUBN/crashpunks-3.json"`;

exports[`7 transformNftUri 2`] = `"https://arweave.net/II4z2ziYyqG7-kWDa98lWGfjxRdYOx9Zdld9P_I_kzE/3.json"`;

exports[`7 transformNftUri 3`] = `"https://api.satoshibles.com/token/btc/25"`;
